tests/crypto/test_signing_keypair.py

```diff
@@ -0,0 +1,37 @@
+import unittest
+import sha3
+import msgpack
+from nkms.crypto.keyring.keys import SigningKeypair
+
+
+class TestSigningKeypair(unittest.TestCase):
+    def setUp(self):
+        self.keypair = SigningKeypair()
+        self.msg = b'this is a test'
+
+    def test_signing(self):
+        msg_digest = sha3.keccak_256(self.msg).digest()
+        signature = self.keypair.sign(msg_digest)
+
+        try:
+            sig = msgpack.loads(signature)
+            self.assertTrue(1, len(sig[0]))     # Check v
+            self.assertTrue(32, len(sig[1]))    # Check r
+            self.assertTrue(32, len(sig[2]))    # Check s
+        except Exception as e:
+            self.fail("Failed to msgpack.load signature: {}".format(e))
+
+    def test_verification(self):
+        msg_digest = sha3.keccak_256(self.msg).digest()
+        signature = self.keypair.sign(msg_digest)
+
+        try:
+            sig = msgpack.loads(signature)
+            self.assertTrue(1, len(sig[0]))     # Check v
+            self.assertTrue(32, len(sig[1]))    # Check r
+            self.assertTrue(32, len(sig[2]))    # Check s
+        except Exception as e:
+            self.fail("Failed to msgpack.load signature: {}".format(e))
```

Comment:
This strikes me as odd - what Exception do we expect here and why?

Reply:
We don't expect any `Exception`. We encapsulate it in a try to check if the `msgpack.loads` fails for any reason. If it does, we can capture the Exception it throws, whatever it may be, and fail the test.

Reply:
Unless there some special reason to do this in the context of crypto, best practice here is to just let the exception raise.